ca-cwds/case-management
# Improves dockerfile

Adds:

 - `.dockerignore` file
 - combines all `apt-get` and install commands
 - no longer re-installs bundler (already in base ruby image)
 - installs npm deps
 - reorders layers for better caching
 - runs the webpack bin script
 - exposes ports
 - adds a REFRESH layer

### How to verify locally

build the image and run it:

```sh
docker build -t cmtest .
docker run -itP -e "RAILS_ENV=test" --rm cmtest /bin/bash
```

then check deps:

```
# ruby -v
ruby 2.4.2p198 (2017-09-14 revision 59899) [x86_64-linux]
# rails -v
Rails 5.1.4
# node -v
v8.7.0
# yarn -v
1.2.0
# bundle -v
Bundler version 1.15.4
# bundle exec rails server
```

then grab the host port assignment with `docker ps` and open in a browser. See that react component renders, no errors raised in browser console, and clean std out in the docker container
